### ISSUE TITLE
Parallel commands

### ISSUE THREAD

**Issue by megagosha** [php-telegram-bot/core#1451]
Any updates on this https://github.com/php-telegram-bot/core/pull/1102#issuecomment-647553643?

My problem is that my bot has multiple conversation commands, and I want to explicitly notify user that older conversation is going to be canceled. GenericCommand only runs if no other command is found. So far I can't find any solution to this. 

Am I missing something here?

**Comment by Hitmare** [comment]
As far as i could see in the example command [GenericmessageCommand.php](https://github.com/php-telegram-bot/example-bot/blob/master/Commands/Conversation/GenericmessageCommand.php) of the Conversation Example, there exists the function to check if an conversation is active, on line 79 `$conversation->exists()` 

you should be able to add this check to your conversation command and give the user the appropriate warning/message

**Comment by megagosha** [comment]
If I understand this correctly GenericMessageCommand is not executed on commands, only on messages. So if you send something like /someConv  GenericCommand will be executed only if SomeConvCommand not found and GenericmessageCommand won't be executed at all.

**Comment by megagosha** [comment]
When should I use this method? If I put it inside a command, it will only be executed after another conversation (different command) has already been cancelled.

**Comment by Hitmare** [comment]
you could make a check within a conversation command , something like this

`
//within the "public function execute(): ServerResponse" function

$message = $this->getMessage();

// If a conversation is busy, execute the conversation command after handling the message.
$conversation = new Conversation(
    $message->getFrom()->getId(),
    $message->getChat()->getId()
);

// Fetch conversation command if it exists and execute it.
if ($conversation->exists()) {
    // return message to user that a conversation already exists and that he needs to cancel the other conversation first
}
//if no conversation exists it will continue the command 

`

**Comment by megagosha** [comment]
If I am reading this correctly, calling conversation constructor cancels other last active conversation https://github.com/php-telegram-bot/core/blob/57a649cfcfe35883165c19942b460ea6b2dfd606/src/Conversation.php#L119C8-L123C1

$message->getText approach should work fine but it requires fetching conversations from db and comparing to current in all conversations I implement.

If GenericCommand were always executed  before specific command,  we could have common logic between all commands.

**Comment by Hitmare** [comment]
you are right. it seems that way. but we can "just" copy the check from the file itself and do something like this

!the code was not tested!
```php


 $conversation = ConversationDB::selectConversation($this->user_id, $this->chat_id, 1);
        if (isset($conversation[0])) {
            //Pick only the first element
            $this->conversation = $conversation[0];

            if ($this->name !== $this->conversation['command']) { //check if the command name in $name matches the command name in the conversation
            //return message that another conversation is active. possible to send the user the command name with $this->conversation['command']
            }
         //continue with the command file if the check above results in a matching command name

        }


```
